### PR TITLE
feat: attachments meta and name

### PR DESCRIPTION
**Pull request by gcandal**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-sdk/issues/282 |

### CODE REVIEW

**Comment by gcandal** [comment]
Not ready

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,15 +9,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
-*
+* Ability to override `name` and `meta` for attachments
```
Comment (joamag) [inline CHANGELOG.md:12]: reference the original issue with `- []()`

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,15 +9,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
 ### Changed
 
 *
 
 ### Fixed
 
-*
+* Multipart encoding extended to classes that implement `toString`
+* Multipart encoding no longer sends an extra newline separator on the last field
```
Comment (joamag) [inline CHANGELOG.md:21]: reference the original issue with `- []()`

File: test/js/base/api.js [hunk 3]
```diff
@@ -383,7 +383,7 @@ describe("RipeAPI", function() {
             const remote = ripe.RipeAPI();
 
             [contentType, body] = remote._encodeMultipart({
-                file: new TextEncoder("utf-8").encode("Hello World")
+                file: "Hello World"
```
[the remark this answers is not in the document]
Comment (joamag) [reply test/js/base/api.js:386]: I'm not sure about this change

[the remark this answers is not in the document]
Comment (joamag) [reply test/js/base/api.js:386]: if we do this we need to rething yonius and many others

**Comment by joamag** [comment]
@gcandal this turned out to be trickier than expected :) let's discuss the multipart/form-data encoding part

File: test/js/base/api.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (joamag) [reply test/js/base/api.js:386]: Just made the change in yonius so we're moving forward with this https://github.com/hivesolutions/yonius/commit/31ce32ad305bfe8d3b999ebd821590f0d67902b8

**Review by joamag — APPROVED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**